Intersect.Server/Entities/ProjectileSpawn.cs

```diff
@@ -176,22 +172,20 @@ public bool ShouldHook(Entity en)
                 return false;
             }
 
-            var entityType = en.GetType();
-
-            if (entityType == typeof(Player))
-            {
-                return ProjectileBase.GrappleHookOptions.Contains(Enums.GrappleOptions.Player);
-            }
-            else if(entityType == typeof(Npc))
-            {
-                return ProjectileBase.GrappleHookOptions.Contains(Enums.GrappleOptions.NPC);
-            }
-            else if(entityType == typeof(Resource))
+            switch(en)
             {
-                return ProjectileBase.GrappleHookOptions.Contains(Enums.GrappleOptions.Resource);
-            }
+                case Player _:
+                    return ProjectileBase.GrappleHookOptions.Contains(Enums.GrappleOptions.Player);
 
-            return false;
+                case Npc _:
+                    return ProjectileBase.GrappleHookOptions.Contains(Enums.GrappleOptions.NPC);
+
+                case Resource _:
+                    return ProjectileBase.GrappleHookOptions.Contains(Enums.GrappleOptions.Resource);
+
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(en));
```

Comment:
I think this should just be `new ArgumentException($"Unsupported entity type {en.GetType().FullName}", nameof(en))` instead of an out-of-range.